internals/secrethub/credential_store.go

```diff
@@ -69,10 +74,13 @@ func (store *credentialConfig) Import() (credentials.Key, error) {
 }
 
 func (store *credentialConfig) getCredentialReader() credentials.Reader {
-	if store.AccountCredential != "" {
-		return credentials.FromString(store.AccountCredential)
+	if store.AccountCredential == "" {
+		return store.configDir.Credential()
+	}
+	if store.credentialFlag.HasEnvarValue() {
+		return credentials.FromEnv("SECRETHUB_CREDENTIAL")
```

Comment:
It feels a bit awkward to have the flag already load the credential from the environment variable and validate it, but then load the credential from envvar again here and not use the already loaded value in `store.AccountCredential`.

We're doing all this to pass the source of the credential, did I get that right? If so, I'm wondering whether we can have the flag itself implement `CredentialSource` interface introduced in the accompanying PR https://github.com/secrethub/secrethub-go/pull/223 and return the flag itself as the credential reader?

The `Source` function of the credential flag could use `HasEnvarValue` to determine what to return.

Reply:
Great idea! I implemented an (manually) tested it and it works well. The code is also much cleaner, thanks :+1: